relay-server/src/services/health_check.rs

```diff
@@ -182,127 +196,74 @@ impl HealthCheckService {
         }
     }
 
-    async fn handle_is_healthy(&self, message: IsHealthy) -> Status {
-        let upstream = self.upstream_relay.clone();
-
+    async fn check_readiness(&mut self, shutdown: &ShutdownHandle) -> Status {
+        // Internal metric that we need to be logged in recurring intervals. This is a form of
+        // health check, but it does not contribute to the status of this service.
         if self.config.relay_mode() == RelayMode::Managed {
-            let fut = upstream.send(IsNetworkOutage);
-            tokio::spawn(async move {
-                if let Ok(is_outage) = fut.await {
-                    metric!(gauge(RelayGauges::NetworkOutage) = u64::from(is_outage));
-                }
-            });
-        }
-
-        if matches!(message, IsHealthy::Liveness) {
-            return Status::Healthy;
+            if let Ok(is_outage) = self.upstream_relay.send(IsNetworkOutage).await {
+                metric!(gauge(RelayGauges::NetworkOutage) = u64::from(is_outage));
+            }
```

Comment:
I just realized that this doesn't actually have an influence on the status.

Reply:
Mh since we wait for the response now, should we also give it a timeout (wrap it in `probe()` and always return `Healthy`)?

Reply:
While I moved this code around, I asked myself the question why we put this in here in the first place. We could equally move this into the upstream service's outage monitor and log changes at the source. I *think* this was done so that we keep updating the value even if nothing changes. Not sure what the right placement is.

Agreed, another probe is nice because it parallelizes.